### PR TITLE
Enable backend threads pre-compilation for orcv2 JIT

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

File: build-scripts/config_common.cmake [hunk 1]
```diff
@@ -86,9 +86,11 @@ endif ()
 
 if (WAMR_BUILD_JIT EQUAL 1)
   if (WAMR_BUILD_AOT EQUAL 1)
+    add_definitions("-DWASM_ENABLE_JIT=1")
```
Comment (lum1n0us) [inline build-scripts/config_common.cmake:89]: Just a personal opinion, it is better to operate `WAMR_BUILD_XXX` variables only in this block. And define `WASM_ENABLE_XXX` from L140.

File: core/iwasm/aot/aot_loader.c [hunk 2]
```diff
@@ -2891,45 +2979,113 @@ aot_load_from_comp_data(AOTCompData *comp_data, AOTCompContext *comp_ctx,
...
-            LLVMOrcLLLazyJITLookup(comp_ctx->orc_jit, &func_addr, func_name);
-#else
-        error = LLVMOrcLLJITLookup(comp_ctx->orc_jit, &func_addr, func_name);
-#endif
-        if (error != LLVMErrorSuccess) {
-            char *err_msg = LLVMGetErrorMessage(error);
-            set_error_buf_v(error_buf, error_buf_size,
-                            "failed to compile orc jit function: %s", err_msg);
-            LLVMDisposeErrorMessage(err_msg);
+    bh_print_time("Begin to lookup jit functions");
```
Comment (lum1n0us) [inline core/iwasm/aot/aot_loader.c:2994]: how about "Start to do compilation in background"

File: core/iwasm/aot/aot_loader.c [hunk 3]
```diff
@@ -2891,45 +2979,113 @@ aot_load_from_comp_data(AOTCompData *comp_data, AOTCompContext *comp_ctx,
...
+            orcjit_stop_compiling = true;
+            for (j = 0; j < i; j++) {
+                os_thread_join(orcjit_threads[j], NULL);
+            }
             goto fail3;
         }
-        module->func_ptrs[i] = (void *)func_addr;
     }
 
+#if WASM_ENABLE_LAZY_JIT == 0
```
Comment (lum1n0us) [inline core/iwasm/aot/aot_loader.c:3075]: guess we don't need `WASM_ENABLE_LAZY_JIT == 0`

File: core/iwasm/compilation/aot_llvm.c [hunk 4]
```diff
@@ -1372,70 +1395,46 @@ orc_jit_create(AOTCompContext *comp_ctx)
         goto fail;
     }
 
-    /* ownership transfer
-     * LLVMOrcJITTargetMachineBuilderRef -> LLVMOrcLLJITBuilderRef
-     */
-#if WASM_ENABLE_LAZY_JIT != 0
-    LLVMOrcLLLazyJITBuilderSetNumCompileThreads(
-        builder, WASM_ORC_JIT_COMPILE_THREAD_NUM);
+    if (num_compile_threads < 1)
```
Comment (lum1n0us) [inline core/iwasm/compilation/aot_llvm.c:1398]: how about 
```
#if WASM_ORC_JIT_COMPILE_THREAD_NUM < 1
#error "WASM_ORC_JIT_COMPILE_THREAD_NUM has to be greater than 0"
```

File: core/iwasm/compilation/aot_orc_extra.cpp [hunk 5]
```diff
@@ -149,10 +182,13 @@ LLVMOrcCreateLLLazyJIT(LLVMOrcLLLazyJITRef *Result,
 
     if (!J) {
         Result = nullptr;
-        return wrap(J.takeError());
+        return NULL;
```
Comment (lum1n0us) [inline core/iwasm/compilation/aot_orc_extra.cpp:185]: `LLVMErrorSuccess` is `0`. `return NULL` will confuse the caller

File: core/iwasm/aot/aot_loader.c (hunk 2 above)
Comment (wenyongh) [reply core/iwasm/aot/aot_loader.c:2994]: I did some test, it reduces port of total lookup time, had better add it to startup as quick as we can.

File: core/iwasm/aot/aot_loader.c (hunk 3 above)
Comment (wenyongh) [reply core/iwasm/aot/aot_loader.c:3053]: If it is removed, the WASM_ENABLE_LAZY_JIT isn't used, do we want to disable the eager mode?

File: core/iwasm/compilation/aot_llvm.c (hunk 4 above)
Comment (wenyongh) [reply core/iwasm/compilation/aot_llvm.c:1398]: Good, done.

File: core/iwasm/compilation/aot_orc_extra.cpp (hunk 5 above)
Comment (wenyongh) [reply core/iwasm/compilation/aot_orc_extra.cpp]: done

**Pull request merged**